src/base/orgs/View/Anchor.svelte

```diff
@@ -0,0 +1,211 @@
+    try {
+      action = Action.Sign;
+      state = State.Signing;
+      const signature = await utils.signSafeTransaction(safe.address, safeTxHash, config);
+
+      state = State.Submitting;
+      await config.safe.client?.confirmTransaction(safeTxHash, signature.data);
+
+      state = State.Success;
+    } catch (err) {
+      console.error(err);
+      state = State.Failed;
+    }
+  };
+</script>
+
+<style>
+  .confirmations {
+    margin-right: 0.5rem;
+  }
+  .table {
+    display: grid;
+    grid-template-columns: 5rem 4fr;
+    grid-gap: 1rem;
+    text-align: left;
+  }
+  .table > *:nth-child(odd) { /* Labels */
+    color: var(--color-secondary);
+  }
+  .table > *:nth-child(even) { /* Values */
+    display: flex;
+    align-items: center;
+    justify-content: left;
+  }
+  .avatars {
+    display: flex;
+  }
+</style>
+
+<span class="confirmations">
+  {#if pending > 0}
+    <strong>{pending}</strong> signature(s) pending
+  {/if}
+</span>
+
+<div class="avatars">
+  {#each project.confirmations as signee}
+    <Avatar inline source={signee} address={signee} glowOnHover />
+  {/each}
+</div>
+
+<!-- Check whether the threshold has been matched or passed -->
+{#if pending <= 0}
+  <button on:click|stopPropagation={() => {
+    action = Action.Execute;
+    state = State.Confirm;
+  }} class="tiny">
+    Execute
+  </button>
+  <!-- Check whether or not we've signed this proposal -->
+{:else if isSigned}
+  <span class="badge safe no-margin">✓ signed</span>
+{:else}
+  <button on:click|stopPropagation={() => {
+    action = Action.Sign;
+    state = State.Confirm;
+    }} class="tiny">
+    Confirm
+  </button>
+{/if}
+
+<!-- We've initiated an action -->
+{#if state !== State.Idle && action === Action.Sign}
+  <Modal floating>
+    <span slot="title">
+      <div>⚓</div>
+      <div>Anchor project</div>
+    </span>
+
+    <span slot="subtitle">
+      {#if state == State.Confirm}
+        <span>Initiate the transaction...</span>
+      {:else if state == State.Signing}
+        <span>Sign the transaction in your wallet...</span>
+      {:else if state == State.Submitting}
+        <span>Transaction is being confirmed...</span>
+      {:else if state == State.Success}
+        <span>Transaction confirmed.</span>
+      {/if}
+    </span>
+
+    <span slot="body">
+      {#if state == State.Confirm}
+        <div class="table">
+          <div>Project</div><code>{project.id}</code>
+          <div>Hash</div><code>{project.anchor.stateHash}</code>
+        </div>
+      {/if}
+    </span>
+
+    <span slot="actions">
+      {#if state == State.Confirm}
+        <button class="primary" on:click={() => confirmAnchor(project.safeTxHash)}>
+          Confirm
+        </button>
+        <button class="text" on:click={close}>
+          Cancel
+        </button>
+      {:else if state == State.Success}
+        <!-- TODO: See if there is a better way to refresh the state in the upper components. -->
```

Comment:
You could use `dispatch` to dispatch an event.

Reply:
Good idea, this would be better :+1: